### ISSUE TITLE
Add more accessibility guidelines to docs/how-to-guides/accessibility.md

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#42224]
## What problem does this address?

In the few years I've been following the Gutenberg project, I've been noticing recurring questions, doubts, request for help, about accessibility patterns and best practices. There's a lot of good information related to accessibility in the history of this project but it's all scattered around in the GitHub issues and pull requests.

Often, contributors ask questions about accessibility issues that have already been answered many times. This is perfectly understandable in a so large project where new contributors join and may not be aware of previous knowledge shared between other contributors. While the accessibility team did an excellent job by publishing extensive information on accessibility on the WordPress Handbook pages, I do believe some best practices should be documented within the Gutenberg documentation.

## What is your proposed solution?

I'd like to propose, as an ongoing task, to progressively extend the content published in `docs/how-to-guides/accessibility.md`. That seems the best place to point contributors to because it's within the project itself. Searching information on the WordPress Handbook is not an easy task and it's pretty time consuming. Instead, `accessibility.md` can be quickly found by contributors while they're working on the project.

Right now, `docs/how-to-guides/accessibility.md` contains very little information and it's only about Landmark Regions. It would be great to add more guidelines, as long as they're short and informative, to help everyone improve accessibility in the Gutenberg project. Some of the topics that can be added would answer the most common questions. More topics can be identified and added over time, when they emerge during the ongoing development of the project.

Some example of topics / questions I noticed in the last few weeks:
- Labelling: what is the best way to label form elements?
- Labelling: what is the impact of hiding labels?
- Labelling: what the `accessible name` is and what its importance is for different kinds of accessibility needs..
- Semantics: how to group logically related form elements and what `fieldset` and `legend` do?
- Screen readers: what screen readers modes are and how they impact keyboard interaction implemented via JavaScript.
- Etc.

**Comment by ryanwelcher** [comment]
@afercia this is a great initiative! Wondering if you would be able to start this off or recommend someone who can?
